MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs

```diff
@@ -379,6 +408,47 @@ protected void SetupWindowRootNavigation(UIViewController viewController, MvxRoo
             return Task.FromResult(true);
         }
 
+        protected virtual Task<bool> ShowPopoverViewController(
+            UIViewController viewController,
+            MvxPopoverPresentationAttribute attribute,
+            MvxViewModelRequest request)
+        {
+            if (PopoverViewController != null)
+                throw new MvxException($"Trying to show View type: {viewController.GetType().Name} as popover, but there is already a popover present!");
+
+            // Content size should be set to a target view controller, not the navigation one
+            if (attribute.PreferredContentSize != default(CGSize))
+            {
+                viewController.PreferredContentSize = attribute.PreferredContentSize;
+            }
+
+            // setup popover based on attribute
+            if (attribute.WrapInNavigationController)
+            {
+                viewController = CreateNavigationController(viewController);
+            }
+
+            // Check if there is a modal already presented first. Otherwise use the topmost view controller.
+            var viewHost = ModalViewControllers.LastOrDefault() ?? _window.RootViewController;
+
+            viewController.ModalPresentationStyle = UIModalPresentationStyle.Popover;
+
+            var presentationController = viewController.PopoverPresentationController;
+            presentationController.PermittedArrowDirections = attribute.PermittedArrowDirections;
+
+            var sourceProvider = Mvx.IoCProvider.Resolve<IMvxPopoverPresentationSourceProvider>();
+            sourceProvider.SetSource(presentationController);
+            presentationController.Delegate = new MvxPopoverDelegate(this);
+
+            viewHost.PresentViewController(
+                viewController,
+                attribute.Animated,
+                null);
```

Comment:
I think we should wait with returning until completion

Reply:
Done. See 67196f4